src/opm/output/eclipse/WellDataSerializers.cpp

```diff
@@ -22,10 +22,40 @@
 #include <opm/output/eclipse/WriteRestartHelpers.hpp>
 #include <vector>
 
+// ----------------------------------------------------------------------------
+std::vector<double>
+Opm::RestartIO::Helpers::
+serialize_SCON(int lookup_step,
+               int ncwmax,
+               int nsconz,
+               const std::vector<const Well*>& sched_wells)
+// ----------------------------------------------------------------------------
+{
+    const size_t well_field_size = ncwmax * nsconz;
+    std::vector<double> data(sched_wells.size() * well_field_size, 0);
+    size_t well_offset = 0;
+    for (const Opm::Well* well : sched_wells) {
+        const auto& completions = well->getCompletions( lookup_step );
+        size_t completion_offset = 0;
+        for (const auto& completion : completions) {
+
+            const size_t offset = well_offset + completion_offset;
+
+            data[ offset + SCON_CF_INDEX ] =
+                completion.getConnectionTransmissibilityFactor();
```

Comment:
Somehow we need to get the value *actually used by the simulator* passed in here; we should **not** even think about calculating this with cell geometries and what not in the output layer. 

There is already functionality in the output for taking various quantities calculated by the simulator as "extra" (see also: https://github.com/OPM/opm-common/pull/391). I guess getting prope `CF` values should be solved something like that, or alternatively we can update the connection factors in the `Schedule` object - but that is ugly.

Reply:
To my understanding, the value at SCON_CF_INDEX should be the transmissibility factor, so this value should be OK here.  The problem is the value at SCON_KH_INDEX, which I for the time being have flagged with UNIMPLEMENTED_VALUE.

Reply:
Ok - I don't have the details in my head, but main point still stands, no trivial values should come from the simulator.